### PR TITLE
MISRA Violation Fixes

### PR DESCRIPTION
**Pull request by bradleysmith23**
*Issue #, if available: N/A

*Description of changes:*
Fix MISRA violations for API's added to library from https://github.com/FreeRTOS/Labs-Project-ota-example-for-aws-iot-core
Tested with unit tests as well as cbmc proofs. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kstribrnAmzn** [comment]
This wasn't mentioned in your PR description - but you verified the MISRA violations are fixed with our standard tool (Coverity).

**Review by kstribrnAmzn — APPROVED** [review]
Great job! The changes are very clean

File: source/jobs.c [hunk 1]
```diff
@@ -346,9 +316,9 @@ JobsStatus_t Jobs_GetTopic( char * buffer,
  * @return JobsSuccess if the sequences are the same;
  * JobsNoMatch otherwise
  */
-static JobsStatus_t strnEq( const char * a,
-                            const char * b,
-                            size_t n )
+static JobsStatus_t strnEquals( const char * a,
```
Comment (AniruddhaKanhere) [inline source/jobs.c:319]: Curious why was this renamed.

File: source/jobs.c [hunk 2]
```diff
@@ -574,27 +544,35 @@ static bool isThingnameTopicMatch( const char * topic,
...
         ( void ) strnAppend( expectedTopicBuffer, &start, TOPIC_BUFFER_SIZE, topicSuffix, topicSuffixLength );
 
         isMatch = ( size_t ) strnlen( expectedTopicBuffer, TOPIC_BUFFER_SIZE ) ==
                   topicLength;
-        isMatch = isMatch && strncmp( expectedTopicBuffer, topic, topicLength ) == 0;
+        isMatch = isMatch && ( strncmp( expectedTopicBuffer, topic, topicLength ) == 0 );
+    }
+    else
+    {
+        /* Empty MISRA body */
```
Comment (AniruddhaKanhere) [inline source/jobs.c:575]: I do not think that only a solitary `if` needs an `else`. I think the rule is that if you have `if-else if...` ladder, then you must have an `else`.

Just an observation, not a blocker.

**Review by AniruddhaKanhere — APPROVED** [review]
Approved!

File: source/jobs.c (hunk 1 above)
Comment (bradleysmith23) [reply source/jobs.c:319]: coreJSON also has a static strnEq() function which was a violation, I forget the exact rule number. Speaking with @kstribrnAmzn, renaming this to be the full name "strnEquals" seemed to be the most straight-forward solution.

File: source/jobs.c (hunk 2 above)
Comment (bradleysmith23) [reply source/jobs.c:575]: Looking at the rule, you are correct. I will be creating another PR in this library soon to add the OTA Job Parser from https://github.com/FreeRTOS/Labs-Project-ota-example-for-aws-iot-core and can address this there if that works.

Comment (AniruddhaKanhere) [reply source/jobs.c:575]: Sounds good. Although we might not need to "fix-this" as this is also not a violation :)
I was just making an observation :)
